saleor/checkout/complete_checkout.py

```diff
@@ -1053,21 +1053,33 @@ def _handle_allocations_of_order_lines(
     )
 
 
-def _handle_checkout_discount(order: "Order", checkout: "Checkout"):
+def _create_order_discount(order: "Order", checkout_info: "CheckoutInfo"):
+    checkout = checkout_info.checkout
     if checkout.discount:
-        # store voucher as a fixed value as it this the simplest solution for now.
-        # This will be solved when we refactor the voucher logic to use .discounts
-        # relations
-
-        order.discounts.create(
-            type=DiscountType.VOUCHER,
-            value_type=DiscountValueType.FIXED,
-            value=checkout.discount.amount,
-            name=checkout.discount_name,
-            translated_name=checkout.translated_discount_name,
-            currency=checkout.currency,
-            amount_value=checkout.discount_amount,
-        )
+        checkout_discount = checkout.discounts.first()
+        if checkout_discount and checkout_discount.type == DiscountType.ORDER_PROMOTION:
+            discount_data = model_to_dict(checkout_discount)
+            discount_data["promotion_rule"] = checkout_discount.promotion_rule
+            del discount_data["checkout"]
+            order.discounts.create(**discount_data)
+
+        if not checkout_discount:
```

Comment:
This could be tricky to understand for someone who is not fully familiar with promotions, and checkout-flow. 
Honestly, I am not sure, why in case of having `not checkout_discount` it means that we should prepare the voucher discount. This should be determined by some kind of the flag, or based on object existence.
Maybe adding `voucher` as an additional parameter `voucher` to `_create_order_discount`, could be helpful here?